
project build/make/
diff --git a/build/make/core/Makefile b/build/make/core/Makefile
index 309a179ee0..942b120abc 100644
--- a/build/make/core/Makefile
+++ b/build/make/core/Makefile
@@ -6699,8 +6699,12 @@ INSTALLED_SYSTEM_QEMU_CONFIG := $(PRODUCT_OUT)/system-qemu-config.txt
 $(INSTALLED_SYSTEM_QEMU_CONFIG): $(INSTALLED_SUPERIMAGE_TARGET) $(INSTALLED_VBMETAIMAGE_TARGET)
 	@echo "$(PRODUCT_OUT)/vbmeta.img vbmeta 1" > $@
 	@echo "$(INSTALLED_SUPERIMAGE_TARGET) super 2" >> $@
+	@echo "$(INSTALLED_USERDATAIMAGE_TARGET) userdata 3" >> $@
+	@echo "$(PRODUCT_OUT)/metadata.img metadata 4" >> $@
 $(INSTALLED_QEMU_SYSTEMIMAGE): $(INSTALLED_VBMETAIMAGE_TARGET) $(MK_COMBINE_QEMU_IMAGE) $(SGDISK_HOST) $(SIMG2IMG) \
     $(INSTALLED_SUPERIMAGE_TARGET) $(INSTALLED_SYSTEM_QEMU_CONFIG)
+	dd if=/dev/zero of="$(PRODUCT_OUT)/metadata.img" bs=4k count=4096 status=none
+	$(HOST_OUT_EXECUTABLES)/mke2fs -t ext4 -q "$(PRODUCT_OUT)/metadata.img"
 	@echo Create system-qemu.img now
 	(export SGDISK=$(SGDISK_HOST) SIMG2IMG=$(SIMG2IMG); \
      $(MK_COMBINE_QEMU_IMAGE) -i $(INSTALLED_SYSTEM_QEMU_CONFIG) -o $@)

project device/generic/goldfish/
diff --git a/device/generic/goldfish/fvp.mk b/device/generic/goldfish/fvp.mk
index 888d9f00..f51b84fd 100644
--- a/device/generic/goldfish/fvp.mk
+++ b/device/generic/goldfish/fvp.mk
@@ -19,6 +19,11 @@ PRODUCT_USE_DYNAMIC_PARTITIONS := true
 PRODUCT_FULL_TREBLE_OVERRIDE := true
 PRODUCT_OTA_ENFORCE_VINTF_KERNEL_REQUIREMENTS := false
 
+#
+# This file is to configure vendor/data partitions of emulator-related products
+#
+$(call inherit-product-if-exists, frameworks/native/build/phone-xhdpi-2048-dalvik-heap.mk)
+
 #
 # All components inherited here go to system image
 #
@@ -42,6 +47,8 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_product.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/media_vendor.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/emulated_storage.mk)
 
+$(call inherit-product, packages/modules/Virtualization/apex/product_packages.mk)
+
 PRODUCT_SOONG_NAMESPACES += device/generic/goldfish
 
 PRODUCT_PACKAGES += \
@@ -90,6 +97,8 @@ PRODUCT_COPY_FILES += \
     device/generic/goldfish/fvpbase/fstab.initrd:$(TARGET_COPY_OUT_RAMDISK)/fstab.qemu \
     device/generic/goldfish/fvpbase/init.fvpbase.rc:$(TARGET_COPY_OUT_VENDOR)/etc/init/hw/init.fvpbase.rc \
     device/generic/goldfish/fvpbase/init.qemu.rc:$(TARGET_COPY_OUT_VENDOR)/etc/init/hw/init.qemu.rc \
+    device/generic/goldfish/fvpbase/mini_network.rc:system/etc/init/mini_network.rc \
+    device/generic/goldfish/fvpbase/mini_network.sh:/system/bin/mini_network.sh \
     device/generic/goldfish/fvpbase/required_images:required_images \
     device/generic/goldfish/fvpbase/ueventd.fvp.rc:$(TARGET_COPY_OUT_VENDOR)/ueventd.rc \
     frameworks/av/services/audiopolicy/config/audio_policy_configuration_generic.xml:$(TARGET_COPY_OUT_VENDOR)/etc/audio_policy_configuration.xml \
diff --git a/device/generic/goldfish/fvpbase/fstab.fvpbase b/device/generic/goldfish/fvpbase/fstab.fvpbase
index 6c62809c..ee52b998 100644
--- a/device/generic/goldfish/fvpbase/fstab.fvpbase
+++ b/device/generic/goldfish/fvpbase/fstab.fvpbase
@@ -4,5 +4,6 @@
 # specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
 system   /system     ext4    ro,barrier=1     wait,logical,first_stage_mount
 vendor   /vendor     ext4    ro,barrier=1     wait,logical,first_stage_mount
-/dev/block/mmcblk0  /data    ext4      noatime,nosuid,nodev,nomblk_io_submit,errors=panic   wait,check,quota
+/dev/block/vda3 /data     ext4 noatime,nosuid,nodev,nomblk_io_submit,errors=panic wait,check,quota
+/dev/block/vda4 /metadata ext4 noatime,nosuid,nodev,nomblk_io_submit,errors=panic wait,check,quota
 /devices/*/block/vde  auto  auto      defaults voldmanaged=sdcard:auto,encryptable=userdata
diff --git a/device/generic/goldfish/fvpbase/init.fvpbase.rc b/device/generic/goldfish/fvpbase/init.fvpbase.rc
index 4d3db00c..1e56a6c4 100644
--- a/device/generic/goldfish/fvpbase/init.fvpbase.rc
+++ b/device/generic/goldfish/fvpbase/init.fvpbase.rc
@@ -1,2 +1,13 @@
 on fs
     mount_all /vendor/etc/fstab.fvpbase
+
+on init
+    # Create UDS structure for base VR services.
+    mkdir /dev/socket/pdx 0775 system system
+    mkdir /dev/socket/pdx/system 0775 system system
+    mkdir /dev/socket/pdx/system/buffer_hub 0775 system system
+    mkdir /dev/socket/pdx/system/performance 0775 system system
+    mkdir /dev/socket/pdx/system/vr 0775 system system
+    mkdir /dev/socket/pdx/system/vr/display 0775 system system
+    mkdir /dev/socket/pdx/system/vr/pose 0775 system system
+    mkdir /dev/socket/pdx/system/vr/sensors 0775 system system

project frameworks/base/
diff --git a/frameworks/base/libs/hwui/JankTracker.cpp b/frameworks/base/libs/hwui/JankTracker.cpp
index 1e5be6c3eed7..58281bcee515 100644
--- a/frameworks/base/libs/hwui/JankTracker.cpp
+++ b/frameworks/base/libs/hwui/JankTracker.cpp
@@ -247,7 +247,7 @@ void JankTracker::finishFrame(FrameInfo& frame, std::unique_ptr<FrameMetricsRepo
             for (size_t i = 0; i < static_cast<size_t>(FrameInfoIndex::NumIndexes); i++) {
                 ss << FrameInfoNames[i] << "=" << frame[i] << ", ";
             }
-            ALOGI("%s", ss.str().c_str());
+            ALOGV("%s", ss.str().c_str());
             // Just so we have something that counts up, the value is largely irrelevant
             ATRACE_INT(ss.str().c_str(), ++sDaveyCount);
         }

project frameworks/native/
diff --git a/frameworks/native/libs/gui/DisplayEventDispatcher.cpp b/frameworks/native/libs/gui/DisplayEventDispatcher.cpp
index dfdce20438..5f082042e2 100644
--- a/frameworks/native/libs/gui/DisplayEventDispatcher.cpp
+++ b/frameworks/native/libs/gui/DisplayEventDispatcher.cpp
@@ -136,7 +136,7 @@ int DisplayEventDispatcher::handleEvent(int, int events, void*) {
         const nsecs_t currentTime = systemTime(SYSTEM_TIME_MONOTONIC);
         const nsecs_t vsyncScheduleDelay = currentTime - mLastScheduleVsyncTime;
         if (vsyncScheduleDelay > WAITING_FOR_VSYNC_TIMEOUT) {
-            ALOGW("Vsync time out! vsyncScheduleDelay=%" PRId64 "ms", ns2ms(vsyncScheduleDelay));
+            ALOGV("Vsync time out! vsyncScheduleDelay=%" PRId64 "ms", ns2ms(vsyncScheduleDelay));
             mWaitingForVsync = false;
             dispatchVsync(currentTime, vsyncDisplayId /* displayId is not used */,
                           ++mLastVsyncCount, vsyncEventData /* empty data */);

project system/core/
diff --git a/system/core/init/init.cpp b/system/core/init/init.cpp
index 29f643ebd..1754427b9 100644
--- a/system/core/init/init.cpp
+++ b/system/core/init/init.cpp
@@ -1034,7 +1034,7 @@ int SecondStageMain(int argc, char** argv) {
             std::chrono::duration<double> waited =
                     std::chrono::steady_clock::now() - Service::exec_service_started();
             if (waited >= kDiagnosticTimeout) {
-                LOG(ERROR) << "Exec service is hung? Waited " << waited.count()
+                LOG(VERBOSE) << "Exec service is hung? Waited " << waited.count()
                            << " without SIGCHLD";
                 if (!dumped_diagnostics) {
                     DumpPidFds("exec service opened: ", Service::exec_service_pid());
diff --git a/system/core/init/service.cpp b/system/core/init/service.cpp
index bd704cf8e..773ae71b3 100644
--- a/system/core/init/service.cpp
+++ b/system/core/init/service.cpp
@@ -508,7 +508,7 @@ void Service::RunService(const std::optional<MountNamespace>& override_mount_nam
     }
 
     if (task_profiles_.size() > 0 && !SetTaskProfiles(getpid(), task_profiles_)) {
-        LOG(ERROR) << "failed to set task profiles";
+        LOG(VERBOSE) << "failed to set task profiles";
     }
 
     // As requested, set our gid, supplemental gids, uid, context, and
diff --git a/system/core/rootdir/init.rc b/system/core/rootdir/init.rc
index cd71aa8aa..1d27ffa3a 100644
--- a/system/core/rootdir/init.rc
+++ b/system/core/rootdir/init.rc
@@ -660,6 +660,8 @@ on property:ro.product.cpu.abilist64=* && property:bootreceiver.enable=1
     write /sys/kernel/tracing/instances/bootreceiver/events/error_report/error_report_end/enable 1
 
 on post-fs-data
+    mkdir /mnt/shared
+    mount 9p FM /mnt/shared FM trans=virtio,version=9p2000.L
 
     mark_post_data
 
