diff --git a/build/make/core/Makefile b/build/make/core/Makefile
index 309a179ee0..942b120abc 100644
--- a/build/make/core/Makefile
+++ b/build/make/core/Makefile
@@ -6699,8 +6699,12 @@ INSTALLED_SYSTEM_QEMU_CONFIG := $(PRODUCT_OUT)/system-qemu-config.txt
 $(INSTALLED_SYSTEM_QEMU_CONFIG): $(INSTALLED_SUPERIMAGE_TARGET) $(INSTALLED_VBMETAIMAGE_TARGET)
 	@echo "$(PRODUCT_OUT)/vbmeta.img vbmeta 1" > $@
 	@echo "$(INSTALLED_SUPERIMAGE_TARGET) super 2" >> $@
+	@echo "$(INSTALLED_USERDATAIMAGE_TARGET) userdata 3" >> $@
+	@echo "$(PRODUCT_OUT)/metadata.img metadata 4" >> $@
 $(INSTALLED_QEMU_SYSTEMIMAGE): $(INSTALLED_VBMETAIMAGE_TARGET) $(MK_COMBINE_QEMU_IMAGE) $(SGDISK_HOST) $(SIMG2IMG) \
     $(INSTALLED_SUPERIMAGE_TARGET) $(INSTALLED_SYSTEM_QEMU_CONFIG)
+	dd if=/dev/zero of="$(PRODUCT_OUT)/metadata.img" bs=4k count=4096 status=none
+	$(HOST_OUT_EXECUTABLES)/mke2fs -t ext4 -q "$(PRODUCT_OUT)/metadata.img"
 	@echo Create system-qemu.img now
 	(export SGDISK=$(SGDISK_HOST) SIMG2IMG=$(SIMG2IMG); \
      $(MK_COMBINE_QEMU_IMAGE) -i $(INSTALLED_SYSTEM_QEMU_CONFIG) -o $@)
diff --git a/device/generic/goldfish/fvp.mk b/device/generic/goldfish/fvp.mk
index 888d9f00..b0756fa2 100644
--- a/device/generic/goldfish/fvp.mk
+++ b/device/generic/goldfish/fvp.mk
@@ -42,6 +42,9 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_product.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/media_vendor.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/emulated_storage.mk)
 
+$(call inherit-product, packages/modules/Virtualization/apex/product_packages.mk)
+$(call inherit-product-if-exists, frameworks/native/build/phone-xhdpi-2048-dalvik-heap.mk)
+
 PRODUCT_SOONG_NAMESPACES += device/generic/goldfish
 
 PRODUCT_PACKAGES += \
@@ -90,6 +93,8 @@ PRODUCT_COPY_FILES += \
     device/generic/goldfish/fvpbase/fstab.initrd:$(TARGET_COPY_OUT_RAMDISK)/fstab.qemu \
     device/generic/goldfish/fvpbase/init.fvpbase.rc:$(TARGET_COPY_OUT_VENDOR)/etc/init/hw/init.fvpbase.rc \
     device/generic/goldfish/fvpbase/init.qemu.rc:$(TARGET_COPY_OUT_VENDOR)/etc/init/hw/init.qemu.rc \
+    device/generic/goldfish/fvpbase/mini_network.rc:system/etc/init/mini_network.rc \
+    device/generic/goldfish/fvpbase/mini_network.sh:/system/bin/mini_network.sh \
     device/generic/goldfish/fvpbase/required_images:required_images \
     device/generic/goldfish/fvpbase/ueventd.fvp.rc:$(TARGET_COPY_OUT_VENDOR)/ueventd.rc \
     frameworks/av/services/audiopolicy/config/audio_policy_configuration_generic.xml:$(TARGET_COPY_OUT_VENDOR)/etc/audio_policy_configuration.xml \
diff --git a/device/generic/goldfish/fvpbase/fstab.fvpbase b/device/generic/goldfish/fvpbase/fstab.fvpbase
index 6c62809c..234270eb 100644
--- a/device/generic/goldfish/fvpbase/fstab.fvpbase
+++ b/device/generic/goldfish/fvpbase/fstab.fvpbase
@@ -4,5 +4,6 @@
 # specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
 system   /system     ext4    ro,barrier=1     wait,logical,first_stage_mount
 vendor   /vendor     ext4    ro,barrier=1     wait,logical,first_stage_mount
-/dev/block/mmcblk0  /data    ext4      noatime,nosuid,nodev,nomblk_io_submit,errors=panic   wait,check,quota
+/dev/block/vda3 /data     ext4 noatime,nosuid,nodev,nomblk_io_submit,errors=panic wait,check,quota
+/dev/block/vda4 /metadata ext4 noatime,nosuid,nodev,nomblk_io_submit,errors=panic wait,check
 /devices/*/block/vde  auto  auto      defaults voldmanaged=sdcard:auto,encryptable=userdata
diff --git a/device/generic/goldfish/fvpbase/init.fvpbase.rc b/device/generic/goldfish/fvpbase/init.fvpbase.rc
index 4d3db00c..d994c864 100644
--- a/device/generic/goldfish/fvpbase/init.fvpbase.rc
+++ b/device/generic/goldfish/fvpbase/init.fvpbase.rc
@@ -1,2 +1,21 @@
 on fs
     mount_all /vendor/etc/fstab.fvpbase
+
+on init
+    # Create UDS structure for base VR services.
+    mkdir /dev/socket/pdx 0775 system system
+    mkdir /dev/socket/pdx/system 0775 system system
+    mkdir /dev/socket/pdx/system/buffer_hub 0775 system system
+    mkdir /dev/socket/pdx/system/performance 0775 system system
+    mkdir /dev/socket/pdx/system/vr 0775 system system
+    mkdir /dev/socket/pdx/system/vr/display 0775 system system
+    mkdir /dev/socket/pdx/system/vr/pose 0775 system system
+    mkdir /dev/socket/pdx/system/vr/sensors 0775 system system
+
+on post-fs-data
+    mkdir /mnt/shared
+    mount 9p FM /mnt/shared FM trans=virtio,version=9p2000.L
+
+on boot
+    setprop ro.boot.hypervisor.protected_vm.supported 1
+    setprop ro.boot.hypervisor.vm.supported 1
