From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Biggers <ebiggers@google.com>
Date: Wed, 12 Aug 2020 14:00:47 -0700
Subject: _____ANNOTATION: security changes_____

Signed-off-by: Eric Biggers <ebiggers@google.com>
Signed-off-by: Lee Jones <joneslee@google.com>
Change-Id: I2f07540d1fbe7203340a3d26e82c6e32af05f46f
---
 ANNOTATIONS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ANNOTATIONS b/ANNOTATIONS
--- a/ANNOTATIONS
+++ b/ANNOTATIONS
@@ -7,3 +7,4 @@ scheduler changes
 uid_sys_stats support
 cpufreq changes
 mm (memory management) changes
+security changes
